int16[] board_row
